novice-r-practice.Rmd

```diff
@@ -0,0 +1,363 @@
+```
+
+*Find length of time between visits by site.*
+
+10. Find sites that experience any stepwise increase in radiation between visits:
+
+```{r rad-increases}
+cleaned %>%
+  dplyr::filter(quantity == "rad") %>%
+  dplyr::group_by(site_id) %>%
+  dplyr::mutate(delta_rad = reading - dplyr::lag(reading)) %>%
+  dplyr::filter(!is.na(delta_rad)) %>%
+  dplyr::summarize(any_increase = any(delta_rad > 0)) %>%
+  dplyr::filter(any_increase)
+```
+
+*Find sites with visits more than one year apart.*
+
+## Creating charts
+
+We will use data on the mass and home range area (HRA) of various species from:
+
+> Tamburello N, Côté IM, Dulvy NK (2015) Data from: Energy and the scaling of animal space use. Dryad Digital Repository.
+> https://doi.org/10.5061/dryad.q5j65
+
+```{r}
+hra <- readr::read_csv(here::here("novice-r-practice", "home-range-database.csv"))
+head(hra)
+```
+
+1. Look at how mass is distributed:
+
+```{r chart-mass}
+ggplot2::ggplot(hra) +
+  ggplot2::geom_histogram(mapping = aes(x = mean.mass.g))
+```
+
+Try again with `log10.mass`:
+
+```{r chart-log-mass}
+ggplot2::ggplot(hra) +
+  ggplot2::geom_histogram(mapping = aes(x = log10.mass))
+```
+
+*Create histograms showing the distribution of home range area using linear and log scales.*
+
+2. Change the visual appearance of a chart:
+
+```{r change-visual}
+ggplot2::ggplot(hra) +
+  ggplot2::geom_histogram(mapping = aes(x = log10.mass), bins = 100) +
+  ggplot2::ggtitle("Frequency of Species Masses") + ggplot2::xlab("Log10 of Mass") + ggplot2::ylab("Number of Species") +
+  ggplot2::theme_minimal()
+```
+
+*Show the distribution of home range areas with a dark background.*
+
+3. Create a scatterplot showing the relationship between mass and home range area:
+
+```{r scatterplot}
+ggplot2::ggplot(hra) +
+  ggplot2::geom_point(mapping = aes(x = log10.mass, y = log10.hra))
+```
+
+*Create a similar scatterplot showing the relationship between the raw values rather than the log values.*
+
+4. Colorize scatterplot points by class:
+
+```{r colorize-scatterplot}
+hra %>%
+  dplyr::mutate(class_fct = as.factor(class)) %>%
+  ggplot2::ggplot() +
+  ggplot2::geom_point(mapping = aes(x = log10.mass, y = log10.hra, color = class_fct), alpha = 0.5)
+```
+
+*Group by order and experiment with different alpha values.*
+
+5. Create a faceted plot:
+
+```{r facet-plot}
+hra %>%
+  dplyr::mutate(class_fct = as.factor(class)) %>%
+  ggplot2::ggplot() +
+  ggplot2::geom_point(mapping = aes(x = log10.mass, y = log10.hra, color = class_fct), alpha = 0.5) +
+  ggplot2::facet_wrap(~class_fct)
+```
+
+*Create a plot faceted by order for just the reptiles.*
+
+6. Fit a linear regression to the logarithmic data for birds:
+
+```{r fit-line}
+hra %>%
+  dplyr::filter(class == "aves") %>%
+  ggplot2::ggplot() +
+  ggplot2::geom_point(mapping = aes(x = log10.mass, y = log10.hra), alpha = 0.5) +
+  ggplot2::geom_smooth(method = lm, mapping = aes(x = log10.mass, y = log10.hra), color = 'red')
```

Comment:
I advocate for `se = FALSE` in almost all situations

Reply:
Happy to add - what's the reason (and why is `se = TRUE` the default)?

Reply:
Ah now you've forced me to articulate it...I've been flip-flopping for an hour!  Here are my thoughts:

Argument against including bands by default:

The bands answer an inferential question, that is, a question beyond the data at hand.  For us to even ask such a question we need some mechanism for connecting the "data we have", to the "data we *could have* had".  Then, for these bands to be appropriate that mechanism needs to match that assumed in their calculation.  Or, in other words, there are a **ton** of assumptions that go into their calculation and we should not treat those lightly.   We might also ask if the band itself, a 95% CI band for the mean response (variable on y-axis) as a function of the explanatory variable (variable on x-axis), is answering the question we are interested in asking.

Argument for including bands by default:

There is some mechanism we can invoke to convince ourselves our data is subject to random variation.  That variation means we should be somewhat uncertain about where the true line is, and some indication of the uncertainty, even if inappropriate, is better than no indication of uncertainty.

I don't have any problems with this exact example, more with assuming the band is meaningful in all cases.

Reply:
thanks - added